diff --git a/README.rst b/README.rst
new file mode 100644
index 0000000..c3feed1
--- /dev/null
+++ b/README.rst
@@ -0,0 +1,11 @@
+This is `xkeyboard-config from FreeDesktop.org`_, with the addition of my own
+custom keyboard layout based on Workman and Programmer Dvorak for use with my
+TEK-227_.  It expects the keyboard to have its builtin “ISO SE,ES,IT” mapping
+enabled. The only differences with the original distribution are two additional
+files: the `keyboard layout description`_ and this README. The `original README`_
+can be found besides this one.
+
+.. _xkeyboard-config from FreeDesktop.org: https://www.freedesktop.org/wiki/Software/XKeyboardConfig
+.. _TEK-227: https://trulyergonomic.com/store/truly-ergonomic-mechanical-ergonomic-keyboard
+.. _keyboard layout description: symbols/custom
+.. _original README: original-readme.txt
diff --git a/README b/original-readme.txt
similarity index 100%
rename from README
rename to original-readme.txt
diff --git a/rules/base.lists.part b/rules/base.lists.part
index a25ee54..d7787a0 100644
--- a/rules/base.lists.part
+++ b/rules/base.lists.part
@@ -57,7 +57,8 @@
               tr \
               us \
               vn \
-              za
+              za \
+	      custom
 
 ! $thinkpads = thinkpad thinkpad60 thinkpadz60
 
diff --git a/rules/base.xml.in b/rules/base.xml.in
index ddd977c..eef0c01 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -5656,6 +5656,18 @@
       </configItem>
       <variantList></variantList>
     </layout>
+    <layout>
+      <configItem>
+        <name>custom</name>
+        <_shortDescription>custom</_shortDescription>
+        <_description>My own custom layout</_description>
+          <languageList>
+            <iso639Id>eng</iso639Id>
+            <iso639Id>ger</iso639Id>
+            <iso639Id>fra</iso639Id>
+        </languageList>
+      </configItem>
+    </layout>
   </layoutList>
   <optionList>
     <group allowMultipleSelection="true">
diff --git a/symbols/Makefile.am b/symbols/Makefile.am
index 828fb98..cf0e2b6 100644
--- a/symbols/Makefile.am
+++ b/symbols/Makefile.am
@@ -34,6 +34,7 @@ terminate tg \
 tj tm tr tw tz \
 ua us uz vn \
 za \
+custom \
 altwin capslock compose ctrl empty eurosign rupeesign group inet \
 keypad kpdl level3 level5 nbsp olpc shift srvr_ctrl typo
 
diff --git a/symbols/custom b/symbols/custom
new file mode 100644
index 0000000..56ed1d3
--- /dev/null
+++ b/symbols/custom
@@ -0,0 +1,34 @@
+Xkb_symbols "basic" {
+
+    include "us(workman)"
+    name[Group1] = "Programming (Personal)";
+
+    // symbols keys, loosely based on programmer dvorak.
+    // Caps Lock can be used to type strings of digits.
+    key <LSGT> { [ numbersign,      grave       ] };
+    key <AE11> { [ percent,         ampersand   ] };
+    key <TLDE> { [ asciitilde,      dollar      ] };
+    key <AE01> { [ exclam,          1           ] };
+    key <AE02> { [ bracketleft,     2           ] };
+    key <AE03> { [ braceleft,       3           ] };
+    key <AE04> { [ braceright,      4           ] };
+    key <AE05> { [ parenleft,       5           ] };
+    key <AE06> { [ equal,           6           ] };
+    key <AE07> { [ asterisk,        7           ] };
+    key <AE08> { [ parenright,      8           ] };
+    key <AE09> { [ plus,            9           ] };
+    key <AE10> { [ bracketright,    0           ] };
+    key <AE12> { [ minus,           underscore  ] };
+    key <BKSL> { [ backslash,       bar         ] };
+    key <AD11> { [ slash,           question    ] };
+    key <AD12> { [ at,              asciicircum ] };
+    key <AD10> { [ colon,           semicolon   ] };
+    key <AC11> { [ quotedbl,        apostrophe  ] };
+
+    // putting the central keys on the TeK to good use
+    replace key <AB10> { [ Tab    ] };
+    replace key <DELE> { [ Escape ] };
+
+    // Workman turns Caps Lock into backspace, undo
+    replace key <CAPS> { [ Caps_Lock ] };
+};
